master/internal/api_experiment.go

```diff
@@ -319,12 +319,17 @@ func (a *apiServer) DeleteExperiment(
 	results, _, err := exputil.DeleteExperiments(ctx, a.m.system,
 		[]int32{req.ExperimentId}, nil)
 
-	if err == nil {
-		if len(results) == 0 {
-			return nil, errors.Errorf("unknown error during delete query.")
-		} else if results[0].Error != nil {
-			return nil, results[0].Error
-		}
+	// report error from the multi-experiment selection code
+	if err != nil {
+		return nil, err
+	}
+
+	// report any error on the individual experiment
+	if len(results) == 0 {
+		return nil, errors.Errorf("unknown error during delete query.")
```

Comment:
```suggestion
		return nil, errors.Errorf("unknown error during delete query.")
```
why not 'tried to delete more than 1 experiment and received 0 results'

Reply:
This is a hopefully-unreached scenario where we delete 1 experiment, running the same code as multi-experiment, and then we should get the result (error or nil).  If we're here then somehow we didn't record an overall error, and we didn't get a pass or fail from this experiment in our results. I'll make this message more descriptive across bulk_actions